runtime/src/bank.rs

```diff
@@ -5165,6 +5166,170 @@ impl Bank {
         self.collect_rent_eagerly(true);
     }
 
+    /// Adds additional necessary accounts to `minimized_account_set`, then removes all other accounts from
+    /// accounts_db
+    pub fn minimize_bank_for_snapshot(
+        &self,
+        minimized_account_set: DashSet<Pubkey>,
+        starting_slot: Slot,
+        ending_slot: Slot,
+    ) {
+        self.get_rent_collection_accounts_between_slots(
+            &minimized_account_set,
+            starting_slot,
+            ending_slot,
+        );
+        self.minimization_add_vote_accounts(&minimized_account_set);
+        self.minimization_add_stake_accounts(&minimized_account_set);
+        self.minimization_add_owner_accounts(&minimized_account_set);
+        self.minimization_add_programdata_accounts(&minimized_account_set);
+
+        info!(
+            "Generated minimized account set with {} accounts for slots {}..={}",
+            minimized_account_set.len(),
+            starting_slot,
+            ending_slot
+        );
+
+        self.accounts()
+            .accounts_db
+            .minimize_accounts_db(self.slot(), &minimized_account_set);
+    }
+
+    // Used to get vote and node pubkeys in `minimized_bank_for_snapshot`
+    fn minimization_add_vote_accounts(&self, minimized_account_set: &DashSet<Pubkey>) {
+        let initial_accounts_len = minimized_account_set.len();
+        let (_, vote_accounts_measure) = measure!(
+            {
+                self.vote_accounts()
+                    .par_iter()
+                    .for_each(|(pubkey, (_stake, vote_account))| {
+                        minimized_account_set.insert(*pubkey);
+                        if let Ok(vote_state) = vote_account.vote_state().as_ref() {
+                            minimized_account_set.insert(vote_state.node_pubkey);
+                        }
+                    });
+            },
+            "get vote accounts"
+        );
+
+        let total_accounts_len = minimized_account_set.len();
+        let new_accounts_len = total_accounts_len - initial_accounts_len;
+        info!("Added {new_accounts_len} for total of {total_accounts_len} accounts. {vote_accounts_measure}");
```

Comment:
Might be nice to uniquify this string for each of these `minimization_add_*()` functions. Ie, add something like "Added {} vote accounts for total ..."

Also, I see you did a mix of `"{value}"` and `"{}, value"` with these log statements. I think the majority of our codebase uses the latter (which isn't necessarily cause to keep using them), but at the very least, maybe use consistent style in the same function.